typewise_alert.test.py

```diff
@@ -1,11 +1,113 @@
+    self.assertTrue(classify_temperature_breach('PASSIVE_COOLING',35) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('PASSIVE_COOLING',25) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('PASSIVE_COOLING',0) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('PASSIVE_COOLING',-0.01) == 'TOO_LOW') 
+    self.assertTrue(classify_temperature_breach('PASSIVE_COOLING',-1) == 'TOO_LOW') 
+    self.assertTrue(classify_temperature_breach('PASSIVE_COOLING',-25) == 'TOO_LOW') 
+
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',100) == 'TOO_HIGH')
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',46) == 'TOO_HIGH')
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',45.1) == 'TOO_HIGH')
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',45) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',25) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',0) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',-0.01) == 'TOO_LOW') 
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',-1) == 'TOO_LOW') 
+    self.assertTrue(classify_temperature_breach('HI_ACTIVE_COOLING',-25) == 'TOO_LOW') 
+
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',100) == 'TOO_HIGH')
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',46) == 'TOO_HIGH')
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',40.1) == 'TOO_HIGH')
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',40) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',25) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',0) == 'NORMAL') 
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',-0.01) == 'TOO_LOW') 
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',-1) == 'TOO_LOW') 
+    self.assertTrue(classify_temperature_breach('MED_ACTIVE_COOLING',-25) == 'TOO_LOW')
+    
+    self.assertTrue(classify_temperature_breach('None',-25) == 'TOO_LOW') 
+    self.assertTrue(classify_temperature_breach('None',25) == 'TOO_HIGH') 
+    self.assertTrue(classify_temperature_breach('None',0) == 'NORMAL') 
+
+  def test_infers_breach_as_per_limits(self):
+    self.assertTrue(infer_breach(20, 50, 100) == 'TOO_LOW')
+    self.assertTrue(infer_breach(2, 3, 5) == 'TOO_LOW')
+    self.assertTrue(infer_breach(4, 1, 10) == 'NORMAL')
+    self.assertTrue(infer_breach(-4, -100, -2) == 'NORMAL')
+    self.assertTrue(infer_breach(0.4, 0.1, 1.0) == 'NORMAL')
+    self.assertTrue(infer_breach(1.2, 0.1, 1.0) == 'TOO_HIGH')
+    self.assertTrue(infer_breach(0.02, 0.1, 1.0) == 'TOO_LOW')
+
+  def test_sender(self):
+    self.assertTrue(send_to_controller('TOO_HIGH') == True) 

```

Comment:
how do you verify if the return value is indeed from a controller or a email target?